src/metabase/db.clj

```diff
@@ -368,10 +368,12 @@
   ;; Establish our 'default' DB Connection
   (create-connection-pool! (jdbc-details db-details))
 
-  ;; Do any custom code-based migrations now that the db structure is up to date
-  ;; NOTE: we use dynamic resolution to prevent circular dependencies
-  (require 'metabase.db.migrations)
-  ((resolve 'metabase.db.migrations/run-all)))
+  ;; Do any custom code-based migrations now that the db structure is up to date.
+  ;; None of the migrations should be ran when Metabase is launched via `load-from-h2`. That's because they will end up doing things like
+  ;; creating duplicate entries for the "magic" groups and permissions entries. They appropriate things will be copied over from the H2 database.
+  (when-not (= config/launch-command :load-from-h2)
```

Comment:
It's not really state because it doesn't change. It's just information about what command was used to launch Metabase.

If we change the name of the command (unlikely) we'll have to grep the codebase anyway to make sure we update the dox and the entrypoint functions. So I'm not worried about hardcoding it